<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting functions for CNV data • cnvPlotteR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><meta property="og:title" content="Plotting functions for CNV data">
<meta property="og:description" content="A group of functions to explore CNV data produced by (or formatted like) CNV-Seq.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">cnvPlotteR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    

<div id="plotting-functions-for-cnv-data" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#plotting-functions-for-cnv-data" class="anchor"></a>Plotting functions for CNV data</h1></div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install from github:</p>
<pre><code>git clone https://github.com/nriddiford/cnvPlotteR.git</code></pre>
<p>Start an R session, and install package:</p>
<pre class="{r}"><code>library(devtools)
install_github("nriddiford/cnvPlotteR")
library(cnvPlotteR)</code></pre>
</div>
<div id="plot-control-freec-normalised-cns" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-control-freec-normalised-cns" class="anchor"></a>Plot Control Freec normalised CNs</h2>
<p>The function <code>freecPlot</code> will plot all CNs for for all chromosomes (if using a non-Drosophila genome the variables <code>chroms</code> and <code>autosomes</code> should be adjusted accordingly). Sub-clonal CNs are highlighted too.</p>
<pre class="{r}"><code>file.names &lt;- dir("data/freec", pattern = ".txt")
for (f in file.names){
  freecPlot(cnv_file=paste("data/freec/",f, sep=''))
}</code></pre>
</div>
<div id="other-plotting-functions-are-for-cnv-seq-output" class="section level2">
<h2 class="hasAnchor">
<a href="#other-plotting-functions-are-for-cnv-seq-output" class="anchor"></a>Other plotting functions are for CNV-Seq output</h2>
<ul>
<li>Designed to read output from <a href="http://tiger.dbs.nus.edu.sg/cnv-seq/doc/manual.pdf">CNV-Seq</a>, although should work for any coverage data with cols:</li>
</ul>
<p>“chromosome” “start” - 1-based window start “end” - 1-based window end “test” - number of reads mapped in window in tumour sample “ref” - number of reads mapped in window in normal sample “position” - (end - start) +1 “log2” - log2 FC for tumour/normal reads per window</p>
<p>Example output from CNV-Seq:</p>
<pre><code>"chromosome"    "start" "end"   "test"  "ref"   "position"      "log2"  "p.value"       "cnv"   "cnv.size"      "cnv.log2"      "cnv.p.value"
"X"     1       10000   50      74      5000    -0.292601981038923      2.85779196291872e-06    0       NA      NA      NA
"X"     5001    15000   50      72      10000   -0.253073616852285      4.12598434147092e-05    0       NA      NA      NA
"X"     10001   20000   2       2       15000   0.272995194815302       7.96882592007325e-06    0       NA      NA      NA
"X"     15001   25000   0       0       20000   NA      NA      0       NA      NA      NA
"X"     20001   30000   0       0       25000   NA      NA      0       NA      NA      NA
"X"     25001   35000   0       0       30000   NA      NA      0       NA      NA      NA
"X"     30001   40000   0       0       35000   NA      NA      0       NA      NA      NA</code></pre>
<div id="plot-all-cnvs-in-specified-directory-for-all-chromosomes" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-all-cnvs-in-specified-directory-for-all-chromosomes" class="anchor"></a>Plot all cnvs in specified directory for all chromosomes</h3>
<ul>
<li>Currently assumes <em>Drosophila</em> genome, and excludes chroms 4 and Y for grid plot. Will look for .cnv files in ‘data/’ unless alternative path provided</li>
</ul>
<pre><code><a href="reference/allPlot.html">allPlot(path = 'path/to/cnv_files/')</a></code></pre>
<p><img src="plots/test_CNVs.png" alt="allPlot example"></p>
</div>
<div id="plot-single-chromosome" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-single-chromosome" class="anchor"></a>Plot single chromosome</h3>
<ul>
<li>Will default to plotting X chromosome if <code>chrom</code> not specified</li>
</ul>
<pre><code><a href="reference/chromPlot.html">chromPlot(cnv_file = "data/test.window-10000.cnv")</a></code></pre>
<p><img src="plots/chroms/test_X_CNVs.png" alt="chromPlot X example"></p>
<ul>
<li>Plot all chroms for all files:</li>
</ul>
<pre class="{r}"><code>file.names &lt;- dir("data/", pattern = ".cnv")

for (f in file.names){
  for (c in c("2L", "2R", "3L", "3R", "X")){
    chromPlot(chrom=c, cnv_file=paste("data/cnvs/",f, sep=''))
  }
}</code></pre>
</div>
<div id="plot-region-for-specified-chromosome" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-region-for-specified-chromosome" class="anchor"></a>Plot region for specified chromosome</h3>
<ul>
<li>Will by default plot region around N on X if no <code>to</code> or <code>from</code> specified</li>
</ul>
<p><code><a href="reference/regionPlot.html">regionPlot(cnv_file="data/w500/test.window-500.cnv", from=3050000, to=3450000, chrom="X", ylim=c(-7,7), bp1=3129368,bp2=3352041, tick=100000, title="222Kb DEL on X")</a></code></p>
<p><img src="plots/regions/test.X_3050000-3450000.png" alt="regionPlot X example"></p>
</div>
<div id="plot-coverage-for-all--count-files" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-coverage-for-all--count-files" class="anchor"></a>Plot coverage for all ‘.count’ files</h3>
<ul>
<li>Use defaults: readLength = 100, windowSize = 50000</li>
</ul>
<pre class="{r}"><code>file.names &lt;- dir("data/counts/", pattern = ".count")

for (f in file.names){
  plotCoverage(counts_file=paste("data/counts/",f, sep=''))
}</code></pre>
<p><img src="plots/coverage/test_coverage.png" alt="plotCoverage example"></p>
</div>
</div>
</div>
<div id="to-do" class="section level1">
<h1 class="hasAnchor">
<a href="#to-do" class="anchor"></a>To do</h1>
<ul>
<li>[x] Allow user to control Y axis limits (for subtle CN changes)</li>
<li>[ ] Build in depth plots</li>
<li>[ ] Fix the ticks in Notch plots (show every 10kb?, start at round number)</li>
<li>[ ] Clean up the function calls (i.e. chrom=NA -&gt; chrom=X to set as default)</li>
</ul>
</div>

  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="NA">NA</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php">MIT</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Nick Riddiford <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://mybinder.org/v2/gh/nriddiford/cnvPlotteR/master"><img src="https://mybinder.org/badge.svg" alt="Binder"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Nick Riddiford.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
